test/powershell/engine/Job/Jobs.Tests.ps1

```diff
@@ -1,23 +1,275 @@
+            $result = Start-Job -Name 'ReceiveJobAutoRemove' -ScriptBlock { 1 + 1 } | Receive-Job -Wait -AutoRemoveJob
+            $result | Should Be 2
+            { Get-Job -Name 'ReceiveJobAutoRemove' -ErrorAction Stop } | ShouldBeErrorId 'JobWithSpecifiedNameNotFound,Microsoft.PowerShell.Commands.GetJobCommand'
+        }
+
+        It 'Can receive-job and keep results' {
+            $job = Start-Job -ScriptBlock { 1 + 1 } | Wait-Job
+            $result = Receive-Job -Keep -Job $job
+            $result | Should Be 2
+            $result2 = Receive-Job -Job $job
+            $result2 | Should Be 2
+            $result3 = Receive-Job -Job $job
+            $result3 | Should BeNullOrEmpty
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job with NoRecurse' {
+            $job = Start-Job -ScriptBlock { 1 + 1 }
+            $result = Receive-Job -Wait -NoRecurse -Job $job
+            $result | Should BeNullOrEmpty
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job using ComputerName' {
+            $jobName = 'ReceiveUsingComputerName'
+            $job = Start-Job -ScriptBlock { 1 + 1 } -Name $jobName | Wait-Job
+            $result = Receive-Job -ComputerName localhost -Job $job
+            $result | Should Be 2
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+
+        It 'Can receive-job using Location' {
+            $jobName = 'ReceiveUsingLocation'
+            $job = Start-Job -ScriptBlock { 1 + 1 } -Name $jobName | Wait-Job
+            $result = Receive-Job -Location localhost -Job $job
+            $result | Should Be 2
+            $job | Remove-Job -Force -ErrorAction SilentlyContinue
+        }
+    }
+
+    Context 'Get-Job tests' {
+        BeforeAll {
+            $getJobTestCases = @(
+                @{ parameters = @{ Name = $startedJob.Name } ; property = 'Name'},
+                @{ parameters = @{ Id = $startedJob.Id } ; property = 'Id'},
+                @{ parameters = @{ InstanceId = $startedJob.InstanceId } ; property = 'InstanceId'},
+                @{ parameters = @{ State = $startedJob.State } ; property = 'State'}
+            )
+
+            $getJobSwitches = @(
+                @{ parameters = @{ Before = $timeAfterStartedJob }; property = '-Before'},
+                @{ parameters = @{ After = $timeBeforeStartedJob }; property = '-After'},
+                @{ parameters = @{ HasMoreData = $true }; property = '-HasMoreData'}
+            )
+
+            $getJobChildJobs = @(
+                @{ parameters = @{ IncludeChildJob = $true }; property = '-IncludeChildJob'},
+                @{ parameters = @{ ChildJobState = 'Completed' }; property = '-ChildJobState'}
+            )
+        }
+
+        BeforeEach {
+            Get-Job | Where-Object { $_.Id -ne $startedJob.Id } | Remove-Job -ErrorAction SilentlyContinue -Force
```

Comment:
I prefer AfterEach for clean up. With BeforeEach, the last It will "leak" a Job that it should clean up.

Reply:
Fixed.